src/main/java/com/hms_networks/sc/cumulocity/data/CConnectorDataMgr.java

```diff
@@ -210,10 +185,156 @@ public static void checkForHistoricalDataAndSend(CConnectorMqttMgr mqttMgr) {
+   * Processes the aggregated data points read from the queue and sends them to the MQTT broker.
+   * (Parameterized map type: {@code Map<Date, List<DataPoint>>})
+   *
+   * @param mqttMgr MQTT manager to send data points on
+   * @param datapointsReadFromQueueMap map of aggregated data points read from the historical data
+   *     queue
+   * @param currentReadTimestampMillis timestamp of the current historical data queue read (in
+   *     millis)
+   * @throws Exception if unable to get the ISO 8601 formatted time stamp for a data point
+   */
+  public static void processDataPointsAndSend(
+      CConnectorMqttMgr mqttMgr, Map datapointsReadFromQueueMap, long currentReadTimestampMillis)
+      throws Exception {
+
+    // Send data via MQTT
+    if (!datapointsReadFromQueueMap.isEmpty()) {
+
+      // Map of child device name to list of CConnectorJsonDataPayload(s)
+      Map childDeviceMessageListMap = new HashMap(); // Map<String, List<CConnectorJsonDataPayload>>
+
+      // List of string data points
+      List stringDataPointsList = new ArrayList(); // List<DataPoint>
+
+      // Iterate over each aggregation period timestamp in the map
+      Iterator datapointsReadFromQueueMapIterator =
+          datapointsReadFromQueueMap
+              .entrySet()
+              .iterator(); // Iterator<Map.Entry<Date, List<DataPoint>>>
+      while (datapointsReadFromQueueMapIterator.hasNext()) {
+        Map.Entry datapointsReadFromQueueMapEntry =
+            (Map.Entry)
+                datapointsReadFromQueueMapIterator.next(); // Map.Entry<Date, List<DataPoint>>
+        Date timestamp = (Date) datapointsReadFromQueueMapEntry.getKey();
+        List datapoints = (List) datapointsReadFromQueueMapEntry.getValue();
+
+        // Create map of child device name to CConnectorJsonDataPayload
+        Map childDeviceMessageMap = new HashMap(); // Map<String, CConnectorJsonDataPayload>
+
+        // Iterate over each data point in the list
+        Iterator datapointsIterator = datapoints.iterator(); // Iterator<DataPoint>
+        while (datapointsIterator.hasNext()) {
+          DataPoint datapoint = (DataPoint) datapointsIterator.next();
+
+          // Get and split tag name of data point
+          CConnectorTagName datapointTagName = new CConnectorTagName(datapoint.getTagName());
+
+          // Get data point information
+          Object value = datapoint.getValueObject();
+          String unit = datapoint.getTagUnit();
+          String childDevice = datapointTagName.getChildDevice();
+          String fragment = datapointTagName.getFragment();
+          String series = datapointTagName.getSeries();
+          Date originalTimestamp = datapoint.getTimeStampAsDate();
+
+          // Remove wrapping quotes if present
+          final char quoteChar = '"';
+          final int firstCharIndex = 0;
+          final int secondCharIndex = 1;
+          final int lastCharIndex = unit.length() - 1;
+          if (unit.length() >= 2
```

Comment:
I prefer to avoid catching exceptions as a form of handling a known condition that we can easily check for otherwise.

I can update the logic here to compare lastCharIndex and secondCharIndex (or similar) and avoid the magic number scenario entirely.

Personally, I think `unit.length() >= 2` is more easily understood than something like
```
int lengthOf2 = 2;
unit.length() >= lengthOf2;
```
Unnecessary and/or redundant mini-variables like `lengthOf2` just tend to make the code more difficult to read and follow at quick glance.